homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,85 @@
+#include "compression.hpp"
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(std::array<std::array<uint8_t, width>, height> arr) {
+    std::vector<std::pair<uint8_t, uint8_t>> vec;
+    uint8_t color = 0; // color of current character
+    uint8_t lastcolor = 0; // color of current character 
+    unsigned int sameColorCount = 0; // sameColorCount counts how many neighbouring characters have the same color 
+    // for loop to iterate through all characters in the array arr, x, y - character coordinates
```

Comment:
Comments are redundant. Name of variable is enough to understands what they are doing :)